File: fd-player/src/examples/bicycle_city_js/lib/gear.js
// gear([0, 0], 4, 12, 0)
return (center, radius, teeth, angleAdvance) => {
  const toothSize = 0.7;
  const rawTeeth = typeof teeth === 'number' && Number.isFinite(teeth) ? teeth : 1;
  const totalTeeth = Math.max(1, Math.ceil(rawTeeth));
  const advance = angleAdvance ?? 0;
  const halfStep = Math.PI / rawTeeth;
  const teethLines = Array.from({ length: totalTeeth }, (_, index) => {
    const angle = index * ((2 * Math.PI) / rawTeeth) + advance;
    const base1 = [
      center[0] + Math.sin(angle - halfStep) * radius,
      center[1] + Math.cos(angle - halfStep) * radius
    ];
    const base2 = [
      center[0] + Math.sin(angle + halfStep) * radius,
      center[1] + Math.cos(angle + halfStep) * radius
    ];
    const apex = [
      center[0] + Math.sin(angle) * (radius + toothSize),
      center[1] + Math.cos(angle) * (radius + toothSize)
    ];
    return [
      { type: 'line', data: { from: base1, to: base2, stroke: '#38bdf8', width: 0.3 } },
      { type: 'line', data: { from: base2, to: apex, stroke: '#38bdf8', width: 0.3 } },
      { type: 'line', data: { from: apex, to: base1, stroke: '#38bdf8', width: 0.3 } }
    ];
  });
  return [
    teethLines,
    {
      type: 'circle',
      data: {
        center,
        radius,
        stroke: '#334155',
        width: 1,
        fill: '#334155'
      }
    }
  ];
};

File: examples/testlib/art/cartoon/machine/parts/gear.fs
// gear([0, 0], 4, 12, 0)
(center, radius, teeth, angleAdvance) =>
{
  n: teeth;
  toothSize: 0.7;
  teethLines:
    Range(0, n) reduce (acc, k) =>
      {
        angle: (k * 2 * math.Pi / n) + (angleAdvance ?? 0);
        halfStep: math.Pi / n;
        base1:
        [
          center[0] + math.Sin(angle - halfStep) * radius,
          center[1] + math.Cos(angle - halfStep) * radius
        ];
        base2:
        [
          center[0] + math.Sin(angle + halfStep) * radius,
          center[1] + math.Cos(angle + halfStep) * radius
        ];
        apex:
        [
          center[0] + math.Sin(angle) * (radius + toothSize),
          center[1] + math.Cos(angle) * (radius + toothSize)
        ];
        eval
          acc
          + [
            { type: "line"; from: base1; to: base2; stroke: "#38bdf8"; width: 0.3; },
            { type: "line"; from: base2; to: apex; stroke: "#38bdf8"; width: 0.3; },
            { type: "line"; from: apex; to: base1; stroke: "#38bdf8"; width: 0.3; }
          ];
      }
    ~ [];

  eval
    teethLines
    + [{ type: "circle"; center; radius; stroke: "#334155"; width: 1; fill: "#334155"; }];
}
